builtin/providers/google/resource_dataflow_job_test.go

```diff
@@ -0,0 +1,70 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataflowJobCreate(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataflowJobDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDataflowJob,
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataflowJobExists(
+						"google_dataflow_job.big_data"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDataflowJobDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_dataflow_job" {
+			continue
+		}
+
+		config := testAccProvider.Meta().(*Config)
+		job, _ := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if job != nil {
+			return fmt.Errorf("Job still present")
+		}
+	}
+
+	return nil
+}
+
+func testAccDataflowJobExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+		config := testAccProvider.Meta().(*Config)
+		_, err := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if err != nil {
+			return fmt.Errorf("Job does not exist")
+		}
+
+		return nil
+	}
+}
+
+var testAccDataflowJob = fmt.Sprintf(`
```

Comment:
@danawillow no problem, I think this is all valuable feedback – I'm coming at this from an application development point of view, so there may also be other concerns from people who use Dataflow for more traditional data processing jobs (periodic or on-demand jobs, for example.)

Your facts are correct, with the clarification that I don't think you can actually _restart_ a job once it's stopped.

I think the reason I wanted to use name rather than id is that in my view you're configuring a logical job to be running – if it's not, it needs to be re-launched, but that would mean a new job id. But to me it's still the same job, with the same uniqueish name.

To answer your questions:

> Do you expect to run each job once? On a schedule? On an as-needed basis?

The specific jobs I'm running are run continuously, as streaming applications. They're not ever supposed to stop unless we're swapping them out for a newer version. They read messages from Pubsub that are written by an HTTP frontend. Imagine an exception tracker that needs to group exceptions by type, time, etc. This needs to happen in real time.

> Do you keep your dataflow configs together with the rest of your terraform configs? If so, what do you expect to happen when you run terraform plan/apply to update a different element?

The Dataflow jobs are defined elsewhere, with a build step that compiles a "template" of the job and pushes it to Google Cloud Storage (https://cloud.google.com/dataflow/docs/templates/creating-templates). A template can be instantiated and parameterized, which is all that Terraform does.

This is mostly to be able to bootstrap a complete environment from scratch, as we'll want many instances for development and testing purposes. Once bootstrapped, we would be updating the jobs directly when we need to re-deploy them, using a [separate in-place update API](https://cloud.google.com/dataflow/pipelines/updating-a-pipeline).

> What are you expecting to need to do in order to relaunch a job? Should you have to explicitly remove it from your config (or run terraform destroy), or should running apply on a config where the job has ended be sufficient?

In my understanding, Terraform is about describing the _intended_ state of the final system, and have Terraform make sure you reach that state. As such, if my TF configuration states that I should have a Dataflow job running, then I would assume that `apply` would re-launch it if it had been manually stopped.